starlite/middleware/session.py

```diff
@@ -0,0 +1,217 @@
+from base64 import b64decode, b64encode
+from os import urandom
+from typing import TYPE_CHECKING, Any, Awaitable, Callable, List, Optional
+
+from orjson import dumps
+from orjson.orjson import loads
+from pydantic import (
+    BaseConfig,
+    BaseModel,
+    SecretBytes,
+    conint,
+    conlist,
+    constr,
+    validator,
+)
+from starlette.datastructures import MutableHeaders
+from starlette.requests import HTTPConnection
+from typing_extensions import Literal
+
+from starlite import MissingDependencyException
+from starlite.datastructures import Cookie
+from starlite.middleware.base import MiddlewareProtocol
+
+try:
+    from cryptography.hazmat.primitives.ciphers.aead import AESGCM
+except ImportError as e:
+    raise MissingDependencyException("cryptography is not installed") from e
+
+if TYPE_CHECKING:
+    from starlette.types import ASGIApp, Message, Receive, Scope, Send
+
+ONE_DAY_IN_SECONDS = 60 * 60 * 24
+NONCE_SIZE = 12
+CHUNK_SIZE = 4096
```

Comment:
This should be less than 4096 to accomodate for the word "session=", "session-1=", "session-2", ...